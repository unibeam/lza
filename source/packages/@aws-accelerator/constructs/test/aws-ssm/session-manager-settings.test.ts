/**
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
 *  with the License. A copy of the License is located at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
 *  and limitations under the License.
 */

import * as cdk from 'aws-cdk-lib';
import { SsmSessionManagerSettings } from '../../index';
import { snapShotTest } from '../snapshot-test';

//import { SynthUtils } from '@aws-cdk/assert';

const testNamePrefix = 'Construct(SsmSessionManagerSettings): ';

//Initialize stack for snapshot test and resource configuration test
const stack = new cdk.Stack();

new SsmSessionManagerSettings(stack, 'SsmSessionManagerSettings', {
  s3BucketName: 'bucketName',
  s3KeyPrefix: 'prefix',
  s3BucketKeyArn: 'arn',
  sendToS3: true,
  sendToCloudWatchLogs: true,
  cloudWatchEncryptionEnabled: true,
  cloudWatchEncryptionKey: new cdk.aws_kms.Key(stack, 'CwKey', {}),
  logRetentionInDays: 3653,
  region: 'us-east-1',
  prefixes: { accelerator: 'AWSAccelerator', ssmLog: 'aws-accelerator' },
  ssmKeyDetails: {
    alias: 'accelerator/sessionmanager-logs/session',
    description: 'AWS Accelerator Session Manager Session Encryption',
  },
});

/**
 * SsmSessionManagerSettings construct test
 */
describe('SsmSessionManagerSettings', () => {
  snapShotTest(testNamePrefix, stack);
});
