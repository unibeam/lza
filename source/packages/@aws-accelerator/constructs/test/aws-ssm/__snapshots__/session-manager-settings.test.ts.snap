// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SsmSessionManagerSettings Construct(SsmSessionManagerSettings):  Snapshot Test 1`] = `
{
  "Resources": {
    "CustomSessionManagerLoggingCustomResourceProviderHandler4FE51699": {
      "DependsOn": [
        "CustomSessionManagerLoggingCustomResourceProviderRole1D8EE686",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "REPLACED-GENERATED-NAME.zip",
        },
        "Handler": "__entrypoint__.handler",
        "MemorySize": 128,
        "Role": {
          "Fn::GetAtt": [
            "CustomSessionManagerLoggingCustomResourceProviderRole1D8EE686",
            "Arn",
          ],
        },
        "Runtime": "nodejs18.x",
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "CustomSessionManagerLoggingCustomResourceProviderLogGroupF4E32979": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "KmsKeyId": {
          "Fn::GetAtt": [
            "CwKeyC5A32F94",
            "Arn",
          ],
        },
        "LogGroupName": {
          "Fn::Join": [
            "",
            [
              "/aws/lambda/",
              {
                "Ref": "CustomSessionManagerLoggingCustomResourceProviderHandler4FE51699",
              },
            ],
          ],
        },
        "RetentionInDays": 3653,
      },
      "Type": "AWS::Logs::LogGroup",
      "UpdateReplacePolicy": "Delete",
    },
    "CustomSessionManagerLoggingCustomResourceProviderRole1D8EE686": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Sub": "arn:\${AWS::Partition}:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
          },
        ],
        "Policies": [
          {
            "PolicyDocument": {
              "Statement": [
                {
                  "Action": [
                    "ssm:DescribeDocument",
                    "ssm:CreateDocument",
                    "ssm:UpdateDocument",
                  ],
                  "Effect": "Allow",
                  "Resource": "*",
                },
              ],
              "Version": "2012-10-17",
            },
            "PolicyName": "Inline",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "CwKeyC5A32F94": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "KeyPolicy": {
          "Statement": [
            {
              "Action": "kms:*",
              "Effect": "Allow",
              "Principal": {
                "AWS": {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::",
                      {
                        "Ref": "AWS::AccountId",
                      },
                      ":root",
                    ],
                  ],
                },
              },
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::KMS::Key",
      "UpdateReplacePolicy": "Retain",
    },
    "SsmSessionManagerSettings24721AC9": {
      "DeletionPolicy": "Delete",
      "DependsOn": [
        "CustomSessionManagerLoggingCustomResourceProviderLogGroupF4E32979",
      ],
      "Properties": {
        "ServiceToken": {
          "Fn::GetAtt": [
            "CustomSessionManagerLoggingCustomResourceProviderHandler4FE51699",
            "Arn",
          ],
        },
        "cloudWatchEncryptionEnabled": true,
        "cloudWatchLogGroupName": {
          "Ref": "SsmSessionManagerSettingsSessionManagerCloudWatchLogGroup15AB5AE0",
        },
        "kmsKeyId": {
          "Ref": "SsmSessionManagerSettingsSessionManagerSessionKey23B7175C",
        },
        "s3BucketName": "bucketName",
        "s3EncryptionEnabled": true,
        "s3KeyPrefix": "prefix",
      },
      "Type": "Custom::SsmSessionManagerSettings",
      "UpdateReplacePolicy": "Delete",
    },
    "SsmSessionManagerSettingsSessionManagerCloudWatchLogGroup15AB5AE0": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "KmsKeyId": {
          "Fn::GetAtt": [
            "CwKeyC5A32F94",
            "Arn",
          ],
        },
        "LogGroupName": "AWSAccelerator-sessionmanager-logs",
        "RetentionInDays": 3653,
      },
      "Type": "AWS::Logs::LogGroup",
      "UpdateReplacePolicy": "Retain",
    },
    "SsmSessionManagerSettingsSessionManagerSessionKey23B7175C": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "Description": "AWS Accelerator Session Manager Session Encryption",
        "EnableKeyRotation": true,
        "KeyPolicy": {
          "Statement": [
            {
              "Action": "kms:*",
              "Effect": "Allow",
              "Principal": {
                "AWS": {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::",
                      {
                        "Ref": "AWS::AccountId",
                      },
                      ":root",
                    ],
                  ],
                },
              },
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::KMS::Key",
      "UpdateReplacePolicy": "Retain",
    },
    "SsmSessionManagerSettingsSessionManagerSessionKeyAlias59E0224E": {
      "Properties": {
        "AliasName": "alias/accelerator/sessionmanager-logs/session",
        "TargetKeyId": {
          "Fn::GetAtt": [
            "SsmSessionManagerSettingsSessionManagerSessionKey23B7175C",
            "Arn",
          ],
        },
      },
      "Type": "AWS::KMS::Alias",
    },
    "SsmSessionManagerSettingsSessionManagerUserKMSPolicyFB96BB42": {
      "Properties": {
        "Description": "",
        "ManagedPolicyName": "AWSAccelerator-SessionManagerUserKMS-us-east-1",
        "Path": "/",
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "kms:Decrypt",
                "kms:GenerateDataKey",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::GetAtt": [
                  "SsmSessionManagerSettingsSessionManagerSessionKey23B7175C",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::ManagedPolicy",
    },
  },
}
`;
